### PR TITLE
Feature/alocare automata cursuri studenti

### PR DESCRIPTION
**Pull request by Lordsib**

### CODE REVIEW

File: src/main/java/org/example/Controller/CourseAllocationController.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        enrollmentService.assignRequiredCoursesToStudentForStudyYear(studyYear);
+
+        return ResponseEntity.ok("Courses assigned successfully for the specified study year");
+    }
+
+    @PostMapping("/assign-courses-automatically/{studentId}/{courseId}")
+    public void assignCoursesToStudentAutomatically(@PathVariable("studentId") int studentId,@PathVariable("courseId") int courseId) {
+        enrollmentService.assignRequiredCoursesToStudentAutomatically(studentId,courseId);
+
+//     /   return ResponseEntity.ok("Courses assigned automatically based on current study year");
```
Comment (Piciorus) [inline src/main/java/org/example/Controller/CourseAllocationController.java]: ?

File: index.html [hunk 2]
```diff
@@ -1,11 +1,100 @@
...
+    <script src="https://unpkg.com/htmx.org@1.6.1/dist/htmx.js"></script>
+    <style>
+        body {
+            background-color: #3d5a80;
+            color: white;
+            margin: 0;
+            font-family: "Arial", sans-serif;
+        }
+
+        .header {
```
Comment (Piciorus) [inline index.html]: add this in a separat css file

File: index.html [hunk 3]
```diff
@@ -1,11 +1,100 @@
...
+        <div class="logo-container">
+            <span class="ubb">UBB</span>
+            <span class="oard">oard</span>
+        </div>
+        <div class="button-container-menu">
+            <button class="button">Membrii</button>
+            <button class="button">Cursuri</button>
+            <button class="button">Contul meu</button>
+        </div>
+    </div>
```
Comment (Piciorus) [inline index.html:21]: add this to a different  html file assing courses?

File: pom.xml [hunk 4]
```diff
@@ -48,12 +48,39 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.33</version>
+        </dependency>
+        <dependency>
```
Comment (Piciorus) [inline pom.xml:68]: sync with dev ... those dependencies exist

**Review by Piciorus — CHANGES_REQUESTED** [review]

**Comment by Piciorus** [comment]
please add a print screen to see how it looks like

**Comment by Lordsib** [comment]
[Course Allocation.pdf](https://github.com/732PC/ProiectColectiv732/files/13562756/Course.Allocation.pdf)

File: index.html (hunk 3 above)
Comment (Lordsib) [reply index.html:21]: I don't understand. I added that to look like the mock up. It's not my job

Comment (Piciorus) [reply index.html:21]: you could move this to a different file and to call it here in index..also you should merge in develop...not in main

**Pull request closed without merging**